homeassistant/components/camera/__init__.py

```diff
@@ -277,6 +277,7 @@ async def write_to_mjpeg_stream(img_bytes: bytes) -> None:
     last_image = None
 
     while True:
+        last_fetch = datetime.now()
```

Comment:
This returns the system realtime clock and interval calculations can be inaccurate due to NTP resets, drifts, or adjusted clock timings. 

IMHO You should rather use `time.monotonic()` see https://docs.python.org/3/library/time.html#time.monotonic which returns more accurate value of seconds.

```
before_clock = time.monotonic()
...
after_clock = time.monotonic()
sleep_time = max(0, interval - (after_clock - before_clock))
if sleep_time > 0:
  await asyncio.sleep(sleep_time)
```

Reply:
Thanks for the suggestions @astromechza, updated the PR.